Add docs for .NET SDK per actor type configuration (#2565)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/actors/howto-actors.md

        options.DrainOngoingCallTimeout = TimeSpan.FromSeconds(60);
        options.DrainRebalancedActors = true;
        options.RemindersStoragePartitions = 7;
        options.ReentrancyConfig = new() { Enabled = false };

        // Add a configuration for a specific actor type.
        // This actor type must have a matching value in the base level 'entities' field. If it does not, the configuration will be ignored.
        // If there is a matching entity, the values here will be used to overwrite any values specified in the root configuration.

>         // In the case of this actor, it will have reentrancy enabled and 'MyActor' will not have reentrancy enabled.

Reviewer: This comment is wrong it seems. I suggest either adding another "defaultActorType" or simply saying that another attribute will not be changed.

Author: @msfussell - I'm not fully sure what's wrong here or what you're suggesting. Can you elaborate a little more?

Reviewer: In the code below shouldn't this line 
`        options.Actors.RegisterActor<ReentrantActor>(typeOptions: new()`
be
`        options.Actors.RegisterActor<MyActor>(typeOptions: new()`

```
// In Startup.cs
public void ConfigureServices(IServiceCollection services)
{
    // Register actor runtime with DI
    services.AddActors(options =>
    {
        // Register actor types and configure actor settings
        options.Actors.RegisterActor<MyActor>();

        // Configure default settings
        options.ActorIdleTimeout = TimeSpan.FromMinutes(60);
        options.ActorScanInterval = TimeSpan.FromSeconds(30);
        options.DrainOngoingCallTimeout = TimeSpan.FromSeconds(60);
        options.DrainRebalancedActors = true;
        options.RemindersStoragePartitions = 7;
        options.ReentrancyConfig = new() { Enabled = false };

        // Add a configuration for a specific actor type.
        // This actor type must have a matching value in the base level 'entities' field. If it does not, the configuration will be ignored.
        // If there is a matching entity, the values here will be used to overwrite any values specified in the root configuration.
        // In the case of this actor, it will have reentrancy enabled and 'MyActor' will not have reentrancy enabled.
        options.Actors.RegisterActor<ReentrantActor>(typeOptions: new()
        {
            ReentrancyConfig = new()
            {
                Enabled = true,
            }
        });
    });

    // Register additional services for use with actors
    services.AddSingleton<BankService>();
}
```

Author: @msfussell - Nope, the actor `MyActor` uses the defaults, listed below it. The actor `ReentrantActor`, gets the configuration specified in `options.Actors.RegisterActor<ReentrantActor>(typeOptions: new()`. 

Given the length of the comment, do you think it becomes confusing? Highlighting the section of the comment that agrees with what I have above:

> and 'MyActor' will not have reentrancy enabled.

Do you think it'd be clearer in a table? Or perhaps outside of the code block entirely?

Reviewer: I see now. I have added this suggestion to the comment (see change). "this" was not clear. 
// In this example `ReentrantActor` has reentrancy enabled and however 'MyActor' will not have reentrancy enabled.

I would still add a few words (or a table) outside of the comment to make this clearer.

>         // In the case of this actor, it will have reentrancy enabled and 'MyActor' will not have reentrancy enabled.

Reviewer: ```suggestion
        // In this example `ReentrantActor` has reentrancy enabled and however 'MyActor' will not have reentrancy enabled.
```

Reviewer 2: For Mark's suggestion:
```suggestion
        // In this example, `ReentrantActor` has reentrancy enabled; however, 'MyActor' will not have reentrancy enabled.
```
